### PR TITLE
Format base64 image before sending to iOS/Android

### PR DESCRIPTION
**Pull request by webb04**

### CODE REVIEW

**Comment by webb04** [comment]
I finally got one sending the file string:

Screen Shot 2019-07-08 at 10 46 25

**Comment by webb04** [comment]
The error message is currently in the mapi html form https://github.com/guardian/mobile-apps-api/pull/1405/files#diff-98ca2b7d4a33f2580cc8377d9ff8d919R72

**Comment by webb04** [comment]
Image uploading now reaching formstack
Screen Shot 2019-07-08 at 14 34 02

File: ArticleTemplates/assets/js/bootstraps/campaign.js [hunk 1]
```diff
@@ -13,6 +13,28 @@ function init() {
...
+                    .toString()
+                    .split(';base64,')[1];
+                // remove data:*/*;base64, from the start of the base64 string
+                res(fileAsBase64);
+            }
+        );
+
+        reader.addEventListener('error', () => {
+            displayFileError(campaign, form);
+            res(null);
```
Comment (alexduf) [inline ArticleTemplates/assets/js/bootstraps/campaign.js]: shouldn't the promise be failed here?

File: ArticleTemplates/assets/js/bootstraps/campaign.js [hunk 2]
```diff
@@ -29,26 +51,44 @@ function initCampaign(campaign) {
...
+                promises.push(data[key]);
+            }
+        }
+
+        Promise.all(promises).then(results => {
+            results.map((result, index) => {
+                data[keys[index]] = result;
+            })
+
+            disableButton(form);
```
Comment (alexduf) [inline ArticleTemplates/assets/js/bootstraps/campaign.js]: shouldn't the button be disabled before we try to read the files asynchronously?

File: ArticleTemplates/assets/js/bootstraps/campaign.js [hunk 3]
```diff
@@ -29,26 +51,44 @@ function initCampaign(campaign) {
...
             }
             
             return o;    
         }, {});
-        disableButton(form);
-        submit(data, campaign, form);
+
+        let promises = [];
+        let keys = [];
+        for (var key in data) {
```
Comment (alexduf) [inline ArticleTemplates/assets/js/bootstraps/campaign.js]: It feels odd to have this extra loop here. Shouldn't the loop that goes through each of the form element produce two arrays, one of field values, one of promises?

**Pull request merged**